### ISSUE TITLE
Add an ability to Mark Orders placed with Store Pickup

### ISSUE THREAD

**Issue by maghamed** [magento/inventory#2030]
To prevent Race Condition we have to have an ability to segregate Orders placed with In Store Pickup delivery method out of other orders. This should be done with newly created ExtensionAttribute

https://github.com/magento-engcom/msi/wiki/Support-of-Store-Pickup-for-Multi-Source-Inventory#Race-condition-between-orders

```
**Note: Maybe makes sense to add extension attribute to Order object and specify Source which should be used to fulfill the order OR we can detect the Source checking whether Shipping address matches the addresses of our Sources.
```

**Comment by toddhainsworth** [comment]
Hi @maghamed,

I'm starting to look at tackling this one! Hoping you can confirm or correct my understanding of the requirements?

From what I understand, we want to give Store Pickup orders a higher priority in the SSA by indicating that is a Store Pickup order and modifying aspect(s) of the SSA to take this into account when fetching stock.

To accomplish this we first need to add an extension-attribute to the order entity that says whether the order was placed using the Store Pickup option. I'm not familiar with the Store Pickup mechanism yet and how it's implemented (possibly because it's in active development? Re: https://github.com/magento-engcom/msi/pull/2036 :smile: ) so I'm wondering if this is indicated by the shipping method used rather than by an extension attribute?

Another extension-attribute should also be added to the order entity that holds the Source that the product(s) should be sourced from (are we going to/have we already handled products that are only available in separate Sources?).

I think this issue might be better tackled after the Store Pickup mechanism has been established, what are your thoughts on this?

**Comment by rodrigowebjump** [comment]
Hi @toddhainsworth,

I will try to help. @maghamed please validate our understanding :)

- The extension attribute must save the source selected by the customer (match shipping address from source vs order is a weak relationship, it can be edited)

- Modification in SSA to prevent Race condition: The change in the SSA is not for just store pickup orders. It should be modified to subtract all store pickup ordered quantities from each source before processing. So you will avoid an unnecessary stock transfer. 
In this example (https://github.com/magento-engcom/msi/wiki/Support-of-Store-Pickup-for-Multi-Source-Inventory#Race-condition-between-orders), it's supposed that Customer 2 selected pickup order from Source A.

**Comment by ishakhsuvarov** [comment]
Merged in #2082
